### PR TITLE
Audio setup and config setup

### PR DESCRIPTION
**Pull request by AnnLe4869**
I have setup some very basic audio that can play audio and explosion sound periodically and jump sound on keyboard event. Will look into how to do the partial sound
I also setup the config such that we can read all config from inside a single json file. like this

```cpp
std::cout << "My name is " << SetupParser::getValue("name") << std::endl;
```

right now I put some example setup in the `common/setup.json` but I image we can do more with this.

### CODE REVIEW

File: client/src/Client.cpp [hunk 1]
```diff
@@ -96,6 +104,12 @@ void clientLoop()
 
         // Swap buffers
         glfwSwapBuffers(sge::window);
+
+        if (i % 1000 == 0) {
+            sound::soundManager->explosionSound();
```
Comment (Alanine42) [inline client/src/Client.cpp:109]: this gave me a heart attack

**Comment by Alanine42** [comment]
thanks for setting up audio and config! this looks good.

I can approve this but looks like some merging is necessary

**Review by Alanine42 — APPROVED** [review]

**Review by realmatthewpeng — APPROVED** [review]
Sounds good to me!

**Review by Metarus — APPROVED** [review]
was missing the changing shader functionality (only i noticed this because only i have a potato potato enough to make it necessary) but i readded that; lgtm

**Review by realmatthewpeng — APPROVED** [review]
Still sounds good to me!

**Review by Alanine42 — APPROVED** [review]
appppproved

**Comment by Alanine42** [comment]
still need one reviewer who hasn't pushed 

@Melody-Ruth @benjxia when ur free

File: CMakeSettings.json [hunk 2]
```diff
@@ -1,27 +0,0 @@
-﻿{
```
Comment (Melody-Ruth) [inline CMakeSettings.json:1]: Why did you delete this file? It's how I'm able to run in both debug and release mode. Of course I can have a local copy, but I think other people were using it as well.

File: client/audios/four_seasons/COPYRIGHT [hunk 3]
```diff
@@ -0,0 +1,7 @@
+by John Harrison with the Wichita State University Chamber Players (https://freemusicarchive.org/music/John_Harrison_with_the_Wichita_State_University_Chamber_Players/The_Four_Seasons_Vivaldi)
```
Comment (Melody-Ruth) [inline client/audios/four_seasons/COPYRIGHT:1]: Thanks for including the citation! We don't want to get in trouble

**Comment by Melody-Ruth** [comment]
It works on my machine! I'll approve it once I understand the cmake changes.

**Review by Melody-Ruth — APPROVED** [review]
Lgtm!

**Review by Metarus — APPROVED** [review]
lgtm

**Review by realmatthewpeng — APPROVED** [review]
Sounds good to me once again!

**Comment by benjxia** [comment]
lgtm

**Review by benjxia — APPROVED** [review]

**Review by Alanine42 — APPROVED** [review]
apppppproveeeeeee

**Review by luoluobuli — APPROVED** [review]
approve

**Pull request closed without merging**